rmgpy/molecule/resonance.py

```diff
@@ -1116,3 +1133,183 @@ def _clar_transformation(mol, aromatic_ring):
+                    bond12.decrement_order()
+                    bond12.decrement_order()
+                    bond23.increment_order()
+                    bond23.increment_order()
+                    bond34.decrement_order()
+                    bond34.decrement_order()
+                    structure = mol.copy(deep=True)
+                    bond12.increment_order()
+                    bond12.increment_order()
+                    bond23.decrement_order()
+                    bond23.decrement_order()
+                    bond34.increment_order()
+                    bond34.increment_order()
+                    try:
+                        structure.update_atomtypes(log_species=False)
+                    except AtomTypeError:
+                        pass
+                    else:
+                        structures.append(structure)
+    return structures
+
+def generate_adsorbate_double_shift_down_resonance_structures(mol):
+    """
+    Generate all of the resonance structures formed by the shift of four electrons from a C#C bond to increase the bond
+    order between two X-C atoms by 2.
+    Example XCXC: [X]C#C[X] <=> [X]#CC#[X]
+    (where '#' denotes a triple bond)
+    """
+    cython.declare(structures=list, paths=list, index=cython.int, structure=Graph)
+    cython.declare(atom=Vertex, atom1=Vertex, atom2=Vertex, atom3=Vertex, atom4=Vertex, bond12=Edge, bond23=Edge, bond34=Edge)
+    cython.declare(v1=Vertex, v2=Vertex)
+
+    structures = []
+    if mol.is_bidentate():
+        for atom in mol.vertices:
+            paths = pathfinder.find_adsorbate_delocalization_paths(atom)
+            for atom1, atom2, atom3, atom4, bond12, bond23, bond34 in paths:
+                if bond23.is_triple() and bond12.is_single() and bond34.is_single():
+                    bond12.increment_order()
+                    bond12.increment_order()
+                    bond23.decrement_order()
+                    bond23.decrement_order()
+                    bond34.increment_order()
+                    bond34.increment_order()
+                    structure = mol.copy(deep=True)
+                    bond12.decrement_order()
+                    bond12.decrement_order()
+                    bond23.increment_order()
+                    bond23.increment_order()
+                    bond34.decrement_order()
+                    bond34.decrement_order()
+                    try:
+                        structure.update_atomtypes(log_species=False)
+                    except AtomTypeError:
+                        pass
+                    else:
+                        structures.append(structure)
+    return structures
+
+def generate_adsorbate_conjugate_resonance_structures(mol):
+    """
+    Generate all of the resonance structures formed by the shift of two electrons in a conjugated pi bond system of
+    a bidentate adsorbate with a bridging C atom in between.
+    Example XCHCHXC: [X]#CC=C[X] <=> [X]=C=CC=[X]
```

Comment:
A base case example would help understanding (and good to also keep the preset example as well)

Reply:
What do you mean? I thought `XCHCHXC: [X]#CC=C[X] <=> [X]=C=CC=[X]` was the example.

Reply:
I think this is resolved